RxSwift/Traits/PrimitiveSequence/PrimitiveSequence+Concurrency.swift

```diff
@@ -111,27 +121,37 @@ public extension PrimitiveSequenceType where Trait == MaybeTrait {
             return try await withTaskCancellationHandler(
                 operation: {
                     try await withCheckedThrowingContinuation { continuation in
-                        var didEmit = false
                         var didResume = false
+                        let lock = RecursiveLock()
```

Comment:
Take this with a grain of salt, as far as I can tell `RecursiveLock` is the to-go tool in this codebase.
- does it have to be recursive?
- can it be replaced with atomic compare-and-swap?

Reply:
Hey, that's a good point. I used `RecursiveTool` precisely because of that, I saw that was the tool being used to synchronize code in the rest of the codebase. I initially thought about that but given that even `AtomicInt` inside RxSwift is backed by a `NSLock`, I didn't want to introduce a different way to synchronize this.
If a maintainer wants me to switch to CAS for this I'm happy to do that.

Reply:
I think changing didResume to AtomicInt and using fetchOr might be better 🙏

Reply:
I don't think so, `AtomicInt` is backed by an `NSLock` so the only thing that will add is making this code less clear.